crates/ruff/src/rules/ruff/rules/f_string_conversion.rs

```diff
@@ -0,0 +1,88 @@
+use ruff_diagnostics::{AlwaysAutofixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::helpers::unparse_expr;
+use ruff_text_size::TextSize;
+use rustpython_parser::ast::{self, Expr, ExprKind};
+
+use crate::checkers::ast::Checker;
+
+#[violation]
+pub struct FStringConversion;
+impl AlwaysAutofixableViolation for FStringConversion {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Use conversion in f-string")
+    }
+
+    fn autofix_title(&self) -> String {
+        "Replace f-string function call with conversion".to_string()
+    }
+}
+
+/// RUF010
+pub(crate) fn f_string_conversion(
+    checker: &mut Checker,
+    expr: &Expr,
+    formatted_value: &Expr,
+    conversion: ast::Int,
+) {
+    // Make sure we're in an f-string
+    if !checker.ctx.in_f_string {
+        return;
+    }
+    // Skip if there's already a conversion
+    if conversion != ast::ConversionFlag::None as u32 {
+        return;
+    }
+
+    if let ExprKind::Call(ast::ExprCall {
+        func,
+        args,
+        keywords,
+    }) = &formatted_value.node
+    {
+        if args.len() != 1 || !keywords.is_empty() {
+            // Can't be a conversion otherwise
+            return;
+        }
+
+        let ExprKind::Name(ast::ExprName { id, .. }) = &func.node else {
+            return;
+        };
+
+        match id.as_str() {
+            "ascii" | "str" | "repr" => {
+                let mut diagnostic = Diagnostic::new(FStringConversion, formatted_value.range());
```

Comment:
I think we must check that the functions resolve to the `builtin` functions and not to user-defined functions.

Reply:
Agreed, I added a check

Reply:
I think you can do `checker.ctx.is_builtin(id)` directly

Reply:
Much better, thanks!